src/Microsoft.AzureHealth.DataServices.Core/Protocol/RawRequestUriBuilder.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AzureHealth.DataServices.Protocol
+{
+    internal class RawRequestUriBuilder : RequestUriBuilder
```

Comment:
Curious - why is this class needed?

Reply:
This class reference I took it from the .net core test cases while i was doing R&D. so it does have some methods that are helpful to append some data to the uri. 
We can use this instead of FhirUriPath.  I have used it in GenericRestClient's  CreateRequest method.